From 560ef6d5ea9a9fc3cd5aac3271a8144e64b92fe3 Mon Sep 17 00:00:00 2001
From: Arun Kumar <arunkumar.ev@adlinktech.com>
Date: Mon, 17 Aug 2020 11:04:53 +0530
Subject: [PATCH 2/7] IMX8QM: Initial LVDS backlight changes

GPIO configuration in dts to enable backlight for
LVDS. These are initial changes, backlight is still
not up after these modifications.

Signed-off-by: Arun Kumar <arunkumar.ev@adlinktech.com>
---
 arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi b/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi
index 8eb4fe7..6cc87b4 100644
--- a/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi
@@ -45,6 +45,9 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_regulators>;
+
 		reg_can01_en: regulator-can01-gen {
 			compatible = "regulator-fixed";
 			regulator-name = "can01-en";
@@ -129,6 +132,15 @@
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 		};
+
+		reg_smarc_lcd_vdd: regulator@130 {
+			compatible = "regulator-fixed";
+			reg = <130>;
+			regulator-name = "smarc_lcd_vdd";
+			gpio = <&gpio1 6 GPIO_ACTIVE_HIGH>; /* SMARC LCD_VDD_EN */
+			enable-active-high;
+			regulator-always-on;
+		};
 	};
 
 	sound: sound {
@@ -157,6 +169,7 @@
 		compatible = "pwm-backlight";
 		pwms = <&lvds0_pwm 0 100000 0>;
 
+		enable-gpios = <&gpio3 20 GPIO_ACTIVE_HIGH>;
 		brightness-levels = < 0  1  2  3  4  5  6  7  8  9
 				     10 11 12 13 14 15 16 17 18 19
 				     20 21 22 23 24 25 26 27 28 29
@@ -584,6 +597,13 @@
 		pinctrl_lvds0_pwm0: lvds0pwm0grp {
 			fsl,pins = <
 				SC_P_LVDS0_GPIO00_LVDS0_PWM0_OUT	0x00000020
+				SC_P_ADC_IN2_LSIO_GPIO3_IO20            0x00000020
+			>;
+		};
+
+		pinctrl_regulators: pinctrl_regulators{
+			fsl,pins = <
+				SC_P_LVDS0_I2C0_SCL_LSIO_GPIO1_IO06		0x00000020
 			>;
 		};
 
-- 
2.10.2

